backend/sources/facebook/connector/src/test/java/co/airy/core/sources/facebook/ChannelsControllerTest.java

```diff
@@ -0,0 +1,230 @@
+        content = webTestHelper.post(
+                "/channels.facebook.explore",
+                objectMapper.writeValueAsString(explorePayload))
+            .andExpect(status().isOk()).andReturn().getResponse().getContentAsString();
+
+        assertThat(content, equalTo("{\"data\":[{\"page_id\":\"7562107744668308\",\"name\":\"facebook app\",\"image_url\":\"facebook app image\",\"connected\":true}]}"));
+
+
+        // Disconnect from facebook channel
+        final DisconnectChannelRequestPayload disconnectRequestPayload = new DisconnectChannelRequestPayload(
+                UUID.fromString(jsonNode.get("id").textValue()));
+        webTestHelper.post(
+                "/channels.facebook.disconnect",
+                objectMapper.writeValueAsString(disconnectRequestPayload))
+            .andExpect(status().isNoContent());
+
+        channels = kafkaTestHelper.consumeValues(1, applicationCommunicationChannels.name());
+        assertThat(ChannelConnectionState.DISCONNECTED, equalTo(channels.get(0).getConnectionState()));
+    }
+
+    private void canInstagramConnect() throws Exception {
+        // Connect to instagram channel
+        final ConnectInstagramRequestPayload connectPayload = this.mockConnectInstagramRequestPayload();
+        final PageWithConnectInfo pageWithConnectInfo = this.mockPageWithConnectInfo();
+
+        doReturn("new-long-live-token-string").when(api).exchangeToLongLivingUserAccessToken(connectPayload.getPageToken());
+        doReturn(pageWithConnectInfo).when(api).getPageForUser(connectPayload.getPageId(), "new-long-live-token-string");
+        doNothing().when(api).connectPageToApp(pageWithConnectInfo.getAccessToken());
+
+
+        final String content = webTestHelper.post(
+                "/channels.instagram.connect",
+                objectMapper.writeValueAsString(connectPayload))
+            .andExpect(status().isOk()).andReturn().getResponse().getContentAsString();
+        final JsonNode jsonNode = new ObjectMapper().readTree(content);
+
+        List<Channel> channels = kafkaTestHelper.consumeValues(1, applicationCommunicationChannels.name());
+        final Channel channel = channels.get(0);
+
+        assertThat("instagram", equalTo(channel.getSource()));
+        assertThat(connectPayload.getAccountId(), equalTo(channel.getSourceChannelId()));
+        assertThat(ChannelConnectionState.CONNECTED, equalTo(channel.getConnectionState()));
+
+        final List<Metadata> metadataList = kafkaTestHelper.consumeValues(1, applicationCommunicationMetadata.name());
+        final Metadata metadata = metadataList.get(0);
+
+        assertThat(MetadataKeys.ChannelKeys.NAME, equalTo(metadata.getKey()));
+        assertThat(connectPayload.getName(), equalTo(metadata.getValue()));
+
+
+        // Disconnect from instagram channel
+        final DisconnectChannelRequestPayload disconnectRequestPayload = new DisconnectChannelRequestPayload(
```

Comment:
One more thing I forgot to mention about our integration tests: We generally avoid using the request payloads from that app packages. Because if we want to test the real thing then the payload definition may also contain an error :) 

So there are two way's we usually construct these payloads:
1. Build an inline string, i.e. `String.format("{\"channel_id\":\"%s\"}", channelId)`
2. [Using jackson](https://github.com/airyhq/airy/blob/280bf7ec36094a0c1935ac5a8a0d3d8ab3bf2087/backend/sources/viber/connector/src/main/java/co/airy/core/sources/viber/services/Api.java#L69)